src/fleaMarket/stall/stall.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import { ClanService } from '../../clan/clan.service';
+import { StallResponse } from './payloads/stallResponse';
+import ServiceError from '../../common/service/basicService/ServiceError';
+import { SEReason } from '../../common/service/basicService/SEReason';
+
+@Injectable()
+export class StallService {
+  constructor(private readonly clanService: ClanService) {}
+
+  /**
+   * Returns the stall by clan id
+   */
+  async readOneByClanId(
+    clanId: string,
+  ): Promise<[StallResponse | null, ServiceError[] | null]> {
+    const [clan, error] = await this.clanService.readOneById(clanId);
+
+    if (!clan || !clan.stall || error) {
```

Comment:
The `clanService.readOneById` uses the basic service method that already returns an specific error if the clan is not found so you could do something like this.
```suggestion
    if (error) {
		return [null, error]
	}
```

In case the stall wouldn't be found it would return a Service Error with NOT_FOUND reason.

Reply:
Yes, the `IServiceReturn` type is a tuple, which always follows the logic:

If everything is ok, the first element in tuple will be the response and second will be `null`

If error occurred, the first element will be `null` and the second array with errors.